### Задача 1

Контрольная работа состоит из пяти вопросов. На каждый вопрос приведено четыре варианта ответа, один из которых правильный. Составьте закон распределения числа правильных ответов при простом угадывании. Найдите M(X), D(X).

Зададим дискретную случайную величину - число правильных ответов, как $X$, которая распределена по биномиальному закону с параметрами n = 5 (число вопросов) и вероятностью угадать ответ $p = \frac{1}{4}$ (вероятность правильного ответа из 4-х возможных вариантов). Исходя из параметра n, возможные значения величины $X$: $x_1 = 0$, $x_2 = 1$, $x_3 = 2$, $x_4 = 3$, $x_5 = 4$, $x_6 = 5$. Посчитаем вероятность для каждого значения по формуле Бернулли:

$$P(X = x) = C_n^x p^x (1 - p)^{n - x} = C_5^x (\frac{1}{4})^{x} (1 - \frac{1}{4})^{5 - x} = C_5^x (\frac{1}{4})^{x} (\frac{3}{4})^{5 - x}$$

In [1]:
from math import factorial

In [2]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

Считаем значения вероятностей для возможный значений $X$:

In [3]:
Px_1 = combinations(5, 0) * (0.25 ** 0) * (0.75 ** 5)
Px_1

0.2373046875

In [4]:
Px_2 = combinations(5, 1) * (0.25 ** 1) * (0.75 ** 4)
Px_2

0.3955078125

In [5]:
Px_3 = combinations(5, 2) * (0.25 ** 2) * (0.75 ** 3)
Px_3

0.263671875

In [6]:
Px_4 = combinations(5, 3) * (0.25 ** 3) * (0.75 ** 2)
Px_4

0.087890625

In [7]:
Px_5 = combinations(5, 4) * (0.25 ** 4) * (0.75 ** 1)
Px_5

0.0146484375

In [8]:
Px_6 = combinations(5, 5) * (0.25 ** 5) * (0.75 ** 0)
Px_6

0.0009765625

Таким образом, закон распределения случайной величины $X$:

<table border="3">
    <tr>
        <th>$x$</th>
        <td>0</td>
        <td>1</td>
        <td>2</td>
        <td>3</td>
        <td>4</td>
        <td>5</td>
    </tr>
    <tr>
        <th>$P(X = x)$</th>
        <td>0.2373046875</td>
        <td>0.3955078125</td>
        <td>0.263671875</td>
        <td>0.087890625</td>
        <td>0.0146484375</td>
        <td>0.0009765625</td>
    </tr>
</table>

Математическое ожидание для биномиально распределенной величины $X$ считаем по формуле: $$M(X) = np = 5 \cdot \frac{1}{4} = 1.25$$

Дисперсию для биномиально распределенной величины $X$ считаем по формуле: $$D(X) = np(1 - p) = 5 \cdot \frac{1}{4} \cdot (1 - \frac{1}{4}) = 5 \cdot \frac{1}{4} \cdot \frac{3}{4} = \frac{15}{16} = 0.9375$$

### Задача 2

Вероятность попадания в цель при одном выстреле равна 0.01. Какова вероятность того, что число попаданий при 200 выстрелах будет не менее 5 и не более 10?

Зададим дискретную случайную величину - число попаданий в цель, как $X$. Так как число выстрелов (n = 200) достаточно велико, а вероятность попадания в цель достаточно мала (p = 0.01), то искомую вероятность можно расчитать по формуле Пуассона с параметром $a = n \cdot p = 200 \cdot 0.01 = 2$.

Поскольку по условию задачи дипазон числа попаданий: не менее 5 и не более 10 - это означает, что крайние значения дипазона должны быть включены в расчет. Таким образом расчет искомой вероятности будет выглядеть следующим образом: $$Px = P(5 \leq X \leq 10) = \displaystyle\sum_{i = 5}^{10} \dfrac{2^{i} e^{-2}}{i!}$$ 

In [9]:
import numpy as np

In [10]:
def poisson_proba(k, a):
    
    return (a ** k) * (np.exp(-a)) / np.math.factorial(k)

In [11]:
Px = sum(poisson_proba(k=i, a=2) for i in range(5, 11))
Px

0.052644709119342686

### Задача 3

В условиях предыдущей задачи сколько выстрелов нужно сделать чтобы быть уверенным с вероятностью 0.9, что хотя бы 10 раз будет совершено попадание?

#### Имеющиеся входные данные:

1. Попадание хотя бы 10 раз - это 10 и более попаданий, таким образом выражение вероятности:

2. $P(X \geq 10) = 0.9$

3. Поскольку суммировать от 10 до бесконечности невозможно, поэтому, используя условие, что сумма вероятностей случайной величины всегда равна $1$, получаем что вероятность можно выразить следующим образом:
$$P(X \geq 10) = 1 - P(X < 10)$$ $$P(X < 10) = 1 - 0.9 = 0.1$$

4. Вероятность попадания в цель при одном выстреле равна: $$p = 0.01$$

Количество выстрелов будем находить путем подбора с использованием формулы Пуассона в примерном дипазоне от 100 до 10000 с шагом 1 (для большей точности):

In [12]:
p = 0.01
for i in range(100, 10001):
    Px = sum(poisson_proba(k=j, a=i * p) for j in range(10))
    if 0 < Px <= 0.1:
        break
print(f" Количество выстрелов: {i}")

 Количество выстрелов: 1421


Если теперь вычислить параметр $a = n \cdot p = 1421 \cdot 0.01 = 14.21$ и подставить в формулу Пуассона для расчета $P(X < 10)$: 

In [13]:
Px_check = sum(poisson_proba(k=i, a=14.21) for i in range(10))
round(Px_check, 2)

0.1

Заданная вероятность $P(X < 10)$ равна вероятности с расчетным числом выстрелов $Px(check)$, таким образом ответ на вопрос задачи:

Количество выстрелов, которые необходимо сделать, чтобы быть уверенным с вероятностью 0.9, что хотя бы 10 раз будет совершено попадание: $$n = 1421$$